lib/services/front.ts

```diff
@@ -37,25 +37,31 @@ export class FrontService extends ServiceScaffold<string> implements ServiceList
 
 	constructor(data: FrontConstructor | FrontListenerConstructor) {
 		super(data);
-		this.session = new Front(data.token);
 		if (data.type === ServiceType.Listener) {
 			const listenerData = <FrontListenerConstructor>data;
-			// This swallows webhook events.  When operating on an entire inbox we use its webhook rule,
-			// but a webhook still requires somewhere to send its webhooks to.
+			this.session = new Front(listenerData.token, listenerData.secret);
+			// This swallows webhook events associated with a channel.
+			// When operating on an entire inbox we use its webhook rule,
+			// but a channel still requires somewhere to send its webhooks to.
 			this.registerHandler('front-dev-null', (_formData, response) => {
 				response.sendStatus(200);
 			});
-			// Create an endpoint for this listener and enqueue events
-			this.registerHandler(listenerData.path || FrontService._serviceName, (formData, response) => {
-				this.queueData({
-					context: formData.body.conversation.id,
-					cookedEvent: {},
-					rawEvent: formData.body,
-					source: FrontService._serviceName,
-					type: formData.body.type,
-				});
-				response.sendStatus(200);
+			this.session.registerEvents({
+				hookPath: `/${listenerData.path || FrontService._serviceName}`,
+				server: this.expressApp,
+			}, (error: Error | null, event?: Event) => {
+				if (!error && event) {
+					this.queueData({
+						context: event.conversation.id,
+						cookedEvent: {},
+						rawEvent: event,
+						source: FrontService._serviceName,
+						type: event.type,
+					});
+				}
```

Comment:
There should be some kind of error handling here, even if it's just `console.error`

Reply:
Done